### PR TITLE
docs: add bidirectional spec-implementation cross-references

### PR DESCRIPTION
**Pull request by cuioss-oliver**
## Summary
- Add bidirectional traceability between 13 specification documents and their implementation/test classes
- Replace inline Java code examples in specs with references to actual implementation files
- Add `@see` Javadoc back-references from implementation and test classes to their specifications
- Add "Implementation Status" / "Referenced Implementation" sections to all spec documents

## Details
Processed all 13 specification documents in `doc/specification/`:
1. **jwt-rest-api.adoc** — Fixed stale "pending" test links, added 11 class references
2. **security.adoc** — Added 10 class references, replaced inline JWKS code
3. **observability.adoc** — Replaced 4 inline Java blocks with implementation references
4. **error-handling.adoc** — Replaced ErrorContext inline code
5. **internationalization.adoc** — Replaced 5 inline i18n code blocks
6. **configuration-static.adoc** — Replaced 2 inline Java blocks
7. **configuration-ui.adoc** — Replaced property dependency and i18n init code
8. **configuration.adoc** — Added Implementation Status section
9. **technical-components.adoc** — Replaced inline relationship code
10. **token-validation.adoc** — Replaced 5 inline Java blocks
11. **end-to-end-testing.adoc, integration-patterns.adoc, testing.adoc** — Added Referenced Implementation sections, replaced 7 inline test examples

## Conventions
- `.adoc` files use relative `link:` paths to implementation files
- Java `@see` tags use absolute GitHub URLs (`https://github.com/cuioss/nifi-extensions/tree/main/doc/specification/...`)
- Classes referenced by multiple specs accumulate multiple `@see` tags
- Conceptual examples (YAML, XML, properties, Docker) kept as-is; only Java code mirroring actual implementation replaced

## Test plan
- [x] `./mvnw -Ppre-commit clean install -DskipTests` passes
- [x] `./mvnw clean install` passes (full build with tests)
- [x] All `link:` paths in `.adoc` files point to existing files
- [x] All `@see` references in Javadoc point to existing spec documents

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: nifi-cuioss-ui/src/test/java/de/cuioss/nifi/ui/service/JwtValidationServiceTest.java [hunk 1]
```diff
@@ -41,6 +41,12 @@
  * - TokenValidationResult data class functionality
  * - Error handling and edge cases
  */
+/**
+ * Tests for {@link JwtValidationService}.
+ *
+ * @see <a href="https://github.com/cuioss/nifi-extensions/tree/main/doc/specification/jwt-rest-api.adoc">JWT REST API Specification</a>
+ * @see <a href="https://github.com/cuioss/nifi-extensions/tree/main/doc/specification/token-validation.adoc">Token Validation Specification</a>
+ */
```
[the remark this answers is not in the document]
Comment (cuioss-oliver) [reply nifi-cuioss-ui/src/test/java/de/cuioss/nifi/ui/service/JwtValidationServiceTest.java]: Fixed in f4647cd — merged the duplicate Javadoc blocks into a single class-level Javadoc block with @see references appended to the existing content.
